generic/sosreport.py

```diff
@@ -0,0 +1,149 @@
+
+    def run_cmd_out(self, cmd):
+        return process.system_output(cmd, shell=True, ignore_status=True, sudo=True)
+
+    def setUp(self):
+        dist = distro.detect()
+        sm = SoftwareManager()
+        if not process.system("sosreport", ignore_status=True, sudo=True):
+            self.log.info("sosreport is installed")
+        elif dist.name == 'Ubuntu':
+            if not sm.install("sosreport"):
+                self.skip("Fail to install sosreport required for this test.")
+        elif dist.name == 'redhat':
+            if not sm.install("sos"):
+                self.skip("Fail to install sos package required for this test.")
+        else:
+            self.skip("sosreport is not supported on this distro "
+                      "or package is not installed")
+
+    def test(self):
+        self.log.info(
+            "===============Executing sosreport tool test===============")
+        directory_name = tempfile.mkdtemp()
+        self.run_cmd("sosreport -h")
+        self.run_cmd("sosreport -l")
+        list = self.params.get('list', default=['--all-logs'])
+        for list_item in list:
+            cmd = "sosreport --batch --tmp-dir=%s %s" % (directory_name, list_item)
+            self.run_cmd(cmd)
+        if self.run_cmd_out("sosreport --batch --tmp-dir=%s -n libraries | "
+                            "grep libraries" % directory_name):
+            self.is_fail += 1
+            self.log.info("--skip-plugins option failed")
+        self.run_cmd("sosreport --batch --tmp-dir=%s -e ntp,numa,snmp" % directory_name)
+        if 'sendmail' not in self.run_cmd_out("sosreport --batch --tmp-dir=%s -e "
+                                              "sendmail | grep sendmail" % directory_name):
+            self.is_fail += 1
+            self.log.info("--enable-plugins option failed")
+        self.run_cmd("sosreport --batch --tmp-dir=%s -o "
+                     "pci,powerpc,procenv,process,processor,kdump" % directory_name)
+        if self.run_cmd_out("sosreport --batch --tmp-dir=%s -o cups | grep kernel" % directory_name):
+            self.is_fail += 1
+            self.log.info("--only-plugins option failed")
+        dir_name = self.run_cmd_out("sosreport --batch --tmp-dir=%s --build | "
+                                    "grep located | cut -d':' -f2" % directory_name).strip()
+        if not os.path.isdir(dir_name):
+            self.is_fail += 1
+            self.log.info("--build option failed")
+        self.run_cmd("sosreport --batch --tmp-dir=%s -v" % directory_name)
+        self.run_cmd("sosreport --batch --tmp-dir=%s --verify" % directory_name)
+        if self.run_cmd_out("sosreport --batch --tmp-dir=%s --quiet -e ntp,numa1" % directory_name):
+            self.is_fail += 1
+            self.log.info("--quiet option failed")
+        self.run_cmd("sosreport --batch --tmp-dir=%s --debug" % directory_name)
+        if 'test123' not in self.run_cmd_out("sosreport --batch --ticket-number=test123  | "
```

Comment:
line31-line35
     def run_cmd(self, cmd):
        self.log.info("executing ============== %s =================" % cmd)
        if process.system(cmd, ignore_status=True, sudo=True):
            self.log.info("%s command failed" % cmd)
            self.is_fail += 1


according to avocado usage this can be
self.log.info("executing ============== %s =================", cmd)
self.log.info("%s command failed", cmd)